client/state/ui/selectors.js

```diff
@@ -6,7 +6,7 @@
  */
 export function getSelectedSite( state ) {
 	if ( ! state.ui.selectedSiteId ) {
-		return null;
+		return false;
```

Comment:
> `null` is the more "semantically accurate" value to return when there's no currently selected site. 

Yeah, I figured that was the reason.

> That `false` results in `undefined` for accessing properties seems like a quirk that we shouldn't be relying on

Fair enough. I can file a PR to roll back, and guard manually inside our section.